signatures/version1.go

```diff
@@ -75,16 +77,36 @@ func Sign(pathToPrivateKey string, signatureData *SignatureData) (string, error)
 // and to be issued by a chain leading to the root CA certificate. There's a
 // util function for this named Validate.
 func Verify(signature string, signatureData *SignatureData) (*x509.Certificate, error) {
+	var signatureBytes []byte
+	var err error
+
 	if signatureData == nil {
 		return nil, errors.New("signatureData must be provided")
 	}
 
-	// Use the CA certificate to verify the signature we've received.
-	signatureBytes, err := base64.URLEncoding.DecodeString(signature)
-	if err != nil {
-		return nil, err
+	// Parse signature format
+	parts := strings.Split(signature, ":")
+
+	switch len(parts) {
+	// Original release using URL-safe encoding and no embedded version
+	case 1:
+		signatureBytes, err = base64.URLEncoding.DecodeString(parts[0])
+		if err != nil {
+			return nil, err
+		}
+	case 2:
+		if parts[0] != "v1" {
+			return nil, fmt.Errorf("invalid signature version %q", parts[0])
```

Comment:
Since we hard code the signature version, it might be nice to surface the version we support in this error message.